inference-engine/src/plugin_api/cpp_interfaces/impl/ie_infer_async_request_internal.hpp

```
@@ -59,6 +59,11 @@ class AsyncInferRequestInternal : public IAsyncInferRequestInternal, public Infe
         StartAsyncImpl();
     };
 
+    std::vector<IMemoryStateInternal::Ptr> QueryState() override {
+        // meaning base plugin reports as no state available - plugin owners need to create proper override of this
+        return {};
```

is not it better to throw an exception?

---

agree, will fix